aws_lambda_powertools/metrics/metric.py

```diff
@@ -1,7 +1,7 @@
 import json
 import logging
 from contextlib import contextmanager
-from typing import Dict, Optional, Union
+from typing import Dict, Iterator, Optional, Union
```

Comment:
See reason at the bottom

```suggestion
from typing import Dict, Optional, Union, Generator
```